### ISSUE TITLE
'Use Cases' section on opendreamkit.org

### ISSUE THREAD

**Issue by nthiery** [OpenDreamKit/OpenDreamKit.github.io#75]
Our web site now has a [`Use cases` section](http://opendreamkit.org/project/use-cases/), currently available from the Project menu. It's meant to hold short documents such as [this one](http://opendreamkit.org/2017/11/02/use-case-publishing-reproducible-notebooks/) with the following structure:
- A user-centric scenario
- A suggestion of solution (or solutions)
- Links to favorite instances; estimate of the number of such instances.
- Time and expertise required
- Links to tutorials, relevant software, etc.
- What's new since ODK started
- ODK's contributions

Such documents can be both useful to our readers (providing them with operational solutions and best practices) and to us (showcasing our work, with a concrete perspective on the impact of our actions). In particular it will highlight the versatility brought by the toolkit approach.

Currently, there is a single use case; we should write more! By design they can be relatively short and fun to write, so please volunteer.

Here are some suggestions:

- [x] Publishing interactive textbooks (@marcinofulus)
- [x] Publishing live talk slides with SageMath + Jupyter + Rise + binder (@nthiery)
- [x] Publishing reproducible logbooks (@nthiery)
- [x] Real time collaboration (@minrk)
- [ ] Getting training (Soft Carpentry, ... @VivianePons, @alex-konovalov)
- [ ] Teaching / training with Jupyter
- [ ] Exploiting modern architecture (@ClementPernet)
- [ ] 3D visualization (@marcinofulus, ...)
- [ ] A group of coauthors wanting to collaborate on code/paper/...
- [ ] Making a software package trivial to try and access (reproducible live documentation and live demos, easy installation)
- [ ] Deploy your own VRE (@defeo?): a lab wanting to provide remote access to its computing resources
    - [ ] Providing (remote) access to (hosted) interactive software to the students in a class 
    - [ ] University/Institution/European Infrastructure wide deployment to provide researchers with easy access to computing resources, software and data
    - [ ] A community wanting to deploy a dedicated VRE (e.g. JOOMMF's VRE) (@fangohr ?)
- [x] WP6 use case: multi-system math calculations with the Math-in-the-Middle (@kohlhase ?)

Any others welcome!

Once we have enough documents, we should also make the use case section very prominent; possibly moving it to the front page.

**Comment by serge-sans-paille** [comment]
Related to the forthcoming workshop, maybe

- [ ] Write high-level code that runs at « native speed », either using a ffi tool or a compiler

**Comment by nthiery** [comment]
>    [ ] Write high-level code that runs at « native speed », either using a
>    ffi tool or a compiler

That's a very good point: we should have use cases about performance,
and in particular the ease of writing fast code.

There remains to frame this into a compelling user story. Something
that makes a visitor reading the scenario immediately think "ah yes, I
have been in those shoes; many people will need that"!

**Comment by olexandr-konovalov** [comment]
A group of coauthors want to collaborate on the GAP code in the GitHub repository, not organising it in a form of GAP package, but using the same as GAP packages for CI and test coverage. Eventually they put a demo of their code in a Jupyter notebook, runnable on Binder, and used [the RISE extension](https://rise.readthedocs.io/) to use the same notebook in a slideshow mode.

This is https://github.com/sukru-yalcinkaya/unipoly described in the Appendix to D5.15.

**Comment by nthiery** [comment]
> A group of coauthors want ...

That sounds very nice! Are you willing to write up a brief use-case
document about it?

**Comment by olexandr-konovalov** [comment]
@nthiery I do. They also have a press-release at https://www.mub.eps.manchester.ac.uk/in-abstract/adjoint-representations-of-black-box-groups/. Collaborative visits were supported by CCP-CoDiMa, and then we agreed that this could be a basis for an interesting demonstration of ODK.

An abstract use case would go further - start the project this way, instead of organising the code after, and submit is as a supplementary material to the paper when it is published.

**Comment by katjabercic** [comment]
We are writing up a use case for [MathHub data](https://data.mathhub.info/).

**Comment by kohlhase** [comment]
and an announcement of the system itself (for 8/2019) that we should have done earlier.

**Comment by kohlhase** [comment]
Katja has written two new posts and committed them, but they are not showing up on the web page (for two hours). 
I wonder whether that is still too busy, ...

**Comment by nthiery** [comment]
They were missing a .md extension. Fixed.

> Katja has written two new posts and committed them, but they are not showing up on the web page (for two hours). 
> I wonder whether that is still too busy, ...

**Comment by kohlhase** [comment]
thanks a lot. I had looked and not seen this. Chapeau

**Comment by nthiery** [comment]
@minrk: thanks for the use case on collaboration!

- What about "asynchronous collaboration" instead of "long term collaboration"?
- About real time collaboration: would there be a tutorial to link to on how to set it up?

**Comment by minrk** [comment]
> What about "asynchronous collaboration" instead of "long term collaboration"?

Great suggestion!

> About real time collaboration: would there be a tutorial to link to on how to set it up?

Yes, I'm working on a demo of this. The Work-in-Progress in JupyterLab is in a bit of a transition, so I'm trying to figure out what will demo best.

**Comment by nthiery** [comment]
Hi @VivianePons, @alex-konovalov,
I just realized that we really should have a use case about "getting training". Pointing to the pages for codima, gap days, sage days, software carpentry, ... A good occasion to highlight odk contribution, special diversity events,  but also that he will last after odk. And to include a couple cool pictures.
Would you be willing to write such a use case?

**Comment by olexandr-konovalov** [comment]
@nthiery @VivianePons good idea - and also about delivering training (for which they can use our training materials).

**Comment by nchauvat** [comment]
(hi)

On Sat, Oct 26, 2019 at 10:15:42AM -0700, Alexander Konovalov wrote:
> @nthiery @VivianePons good idea - and also about delivering training (for which they can use our training materials).

And our jupyter-training extension.

-- 
Nicolas Chauvat

logilab.fr - services en informatique scientifique et gestion de connaissances

**Comment by nthiery** [comment]
> > @nthiery @VivianePons good idea - and also about delivering training (for which they can use our training materials).
> 
> And our jupyter-training extension.

Yes; so really two different use cases:

- Getting training
- Teaching with Jupyter

I added both to my wish list in the first comment of this issue :-)
